<launch>

    <arg name="markerId"        default="26"/>
    <arg name="markerSize"      default="0.08"/>    <!-- in m -->
    <arg name="marker_frame"    default="aruco_marker_frame"/>
    <arg name="camera_name"     default="d_bot_camera"/>
    <arg name="ref_frame"       default=""/>  <!-- leave empty and the pose
    will be published wrt param parent_name -->
    <arg name="corner_refinement" default="LINES" /> <!-- NONE, HARRIS, LINES, SUBPIX -->


    <!-- <node pkg="aruco_ros" type="single" name="aruco_single">
        <remap from="/camera_info" to="/$(arg camera_name)/rgb/camera_info" />
        <remap from="/image"  to="/$(arg camera_name)/rgb/image_raw" />
        <param name="image_is_rectified" value="False"/>
        <param name="marker_size"        value="$(arg markerSize)"/>
        <param name="marker_id"          value="$(arg markerId)"/>
        <param name="reference_frame"    value="$(arg ref_frame)"/>
        <param name="camera_frame"       value="/d_bot_camera_link"/>
        <param name="marker_frame"       value="$(arg marker_frame)" />
        <param name="corner_refinement"  value="$(arg corner_refinement)" />
    </node> -->

    <node pkg="aruco_ros" type="marker_publisher" name="aruco_marker_publisher">
        <remap from="/camera_info" to="/$(arg camera_name)/rgb/camera_info" />
        <remap from="/image" to="/$(arg camera_name)/rgb/image_raw" />
        <param name="image_is_rectified" value="False"/>
        <param name="marker_size"        value="$(arg markerSize)"/>
        <param name="reference_frame"    value="world"/>
        <param name="camera_frame"       value="d_bot_camera_link"/>
    </node>


</launch>

